docs/Workflow.md

```diff
@@ -1,32 +1,55 @@
 <!--- SPDX-License-Identifier: Apache-2.0 -->
 # Contribution Guide
 
-## Prerequisite: Install LLVM
+## Step 1: Fork ONNX-MLIR in the cloud
 
-Follow directions in this section of the top level [README](../README.md#mlir) 
-for installing the currently supported LLVM version. We assume here a
-LINUX installation. 
+We strongly encourage contributors to work in their own forks of the ONNX-MLIR project.
+Creating a new fork is easy.
 
-### Step 1: Fork in the cloud
-
-1. Visit https://github.com/onnx/onnx-mlir
+1. Visit https://github.com/onnx/onnx-mlir.
 2. Click `Fork` button (top right) to establish a fork.
 
-### Step 2: Clone fork to local storage
+For the rest of this guide, we assume that you have set `user` to match your github profile name:
+
+```sh
+user={your github profile name}
+```
+
+## Step 2: Setup MLIR 
+
+### Setup using Docker
+
+Follow the directions provided [here](Docker.md#building-onnx-mlir-in-a-docker-environment) to establish a docker image that uses your ONNX-MLIR fork.
+Make sure to uncomment the additional commands in Step 5 of the `Dockerfile` listed there and replace the `<<GitID>>` string with your own GitHub ID.
+
```

Comment:
At any point you can interact with your Docker image interactively:

```
docker run -it <<ImageName>> /bin/bash
```

Once inside the image you can navigate to the ONNX-MLIR GitHub repository:
```
cd /workdir/onnx-mlir
```

From there you can interact with the repository via the usual Git commands as usual.

Never commit anything to your fork's main branch, the only way you should update `main` is from `upstream`. The procedure to update your fork's `main` is listed below.

To keep your ONNX-MLIR fork's `main` up to date perform the following steps:

1. Fetch the latest versions of your fork (`origin`) and the `upstream` repositories:
```
git fetch --all
```

2. Update the `main` branch on your fork:
```
git checkout main
git merge upstream/main
git push origin main
```

3. The `main` branch of your fork should now be identical to the `main` branch of `upstream`. To check you can do:
```
git diff upstream/main
```
and the command will return immediately signaling that no differences exist between `upstream/main` and `origin/main`

4. (Optional) To update any branches you are currently working to use the latest ONNX-MLIR, you need to do the following:

```
git checkout <my_branch>
git merge origin/main
```

If no conflicts are signaled and the merge is complete do:

```
git push origin <my_branch>
```

However, if conflicts appear, the merge will be interrupted until the conflicts are resolved. A list of files will be marked as containing conflicts. To identify those files do:

```
git status -uno
```

The files in red are the files containing conflicts. Go to all the files which contain a conflict and resolve the conflicts.
When the conflicts are resolved do a `git add` on each conflicted file:

```
git add path/to/file1
git add path/to/file2
...
```

When all conflicted files have been added do:
```
git commit
```

Followed by a git push:
```
git push origin <my_branch>
```